docs/reference/xpkg.md

```diff
@@ -0,0 +1,201 @@
+other layers with the `io.crossplane.xpkg` key are used to signify to
+third-party consumers that a layer contains content related to the `xpkg` that
+may be specific to a given consumer.
+
+If no layer descriptors have an annotation in the form `io.crossplane.xpkg:
+base`, the resultant filesystem from [applying changesets] from all layers will
+be used.
+
+> Crossplane prefers the usage of layers to define additive package content as
+> it provides a clean mechanism to build an `xpkg` through a series of stages. A
+> valid `xpkg` can be produced and later modified while verifying that the
+> integrity of the existing content is not violated, which ensures that
+> Crossplane's package manager will process the resulting `xpkg` in the same
+> manner as the it would prior to modification.
+
+### Configuration
+
+Crossplane imposes no additional requirements on image configuration and does
+not consider its contents when processing a package.
+
+### Layers
+
+As described above, Crossplane is only concerned with the single layer
+referenced by the descriptor containing `io.crossplane.xpkg: base` if
+distinguished. Crossplane imposes no additional restrictions on any other
+layers, including those with a `io.crossplane.xpkg` annotation but a value other
+than `base`, but does require the following of the `xpkg` base layer:
+
+- A single file with name `package.yaml` MUST exist in the root directory of the
+  `xpkg` base layer if distinguished, or in the root of the image filesystem
+  after all layer changesets are applied.
+- The `package.yaml` file MUST contain a valid [YAML stream].
+- All other content in either the `xpkg` base layer, or the full image
+  filesystem is ignored by Crossplane.
+
+> The ability to use the image's flattened filesystem is primarily for backwards
+> compatibility and is not encouraged, especially in the event that an image
+> contains more than just `xpkg` related content, due to the fact that
+> accidentally overwriting or modifying the `xpkg` layer contents in subsequent
+> layers when constructing an image could cause the package to be invalid.
+
+## package.yaml Contents
+
+Depending on the type of package, the YAML stream in the `xpkg` base layer
+`package.yaml` may contain different content. Additionally, the objects in the
+YAML stream may contain common annotations that are suitable for the given
+object type.
+
+### Configuration Package Requirements
+
+The `package.yaml` for Configuration packages must adhere to the following
+requirements:
+
+- One (1) and only one `Configuration.meta.pkg.crossplane.io` object MUST be
+  defined in the YAML stream.
+- Zero (0) or more `CompositeResourceDefinition.apiextensions.crossplane.io`
+  objects MAY be defined in the YAML stream.
+- Zero (0) or more `Composition.apiextensions.crossplane.io` objects MAY be
+  defined in the YAML stream.
+- Zero (0) other object types may be defined in the YAML stream.
```

Comment:
Is there a use case in mind that we'll need to allow arbitrary types in a configuration package?

Reply:
There is not today, so I would opt to explicitly disallow it until we have a use case.